R/deprecated.R

```diff
@@ -1,44 +1,44 @@
+
 #' Print message for deprecated functions in Shiny
 #'
 #' To disable these messages, use `options(shiny.deprecation.messages=FALSE)`.
 #'
-#' @param new Name of replacement function.
-#' @param msg Message to print. If used, this will override the default message.
-#' @param old Name of deprecated function.
-#' @param version The last version of Shiny before the item was deprecated.
+#' @param version Shiny version when the function was deprecated
+#' @param what Function with possible arguments
+#' @param with Possible function with arguments that should be used instead
+#' @param details Additional information to be added after a new line to the displayed message
 #' @keywords internal
-shinyDeprecated <- function(new=NULL, msg=NULL,
-                            old=as.character(sys.call(sys.parent()))[1L],
-                            version = NULL) {
-
-  if (getOption("shiny.deprecation.messages") %||% TRUE == FALSE)
+shinyDeprecated <- function(
+  version, what, with = NULL, details = NULL
+) {
+  if (is_false(getOption("shiny.deprecation.messages"))) {
     return(invisible())
-
-  if (is.null(msg)) {
-    msg <- paste(old, "is deprecated.")
-    if (!is.null(new)) {
-      msg <- paste(msg, "Please use", new, "instead.",
-        "To disable this message, run options(shiny.deprecation.messages=FALSE)")
-    }
   }
 
-  if (!is.null(version)) {
-    msg <- paste0(msg, " (Last used in version ", version, ")")
+  msg <- paste0("`", what, "` is deprecated as of shiny ", version, ".")
+  if (!is.null(with)) {
+    msg <- paste0(msg, "\n", "Please use `", with, "` instead.")
+  }
+  if (!is.null(details)) {
+    msg <- paste0(msg, "\n", details)
   }
 
-  # Similar to .Deprecated(), but print a message instead of warning
-  message(msg)
+  # lifecycle::deprecate_soft(when, what, with = with, details = details, id = id, env = env)
+  rlang::inform(message = msg, .frequency = "always", .frequency_id = msg)
```

Comment:
It would be nice if the output were sent to `stderr()` (with `inform(.file = stderr())`, so that it comes out red in RStudio. This is what it looks like now:

![image](https://user-images.githubusercontent.com/86978/102109638-e4d13900-3df9-11eb-8737-6474e10d4114.png)

Although maybe there's some reason that `inform()` defaults to stdout?

Reply:
In our case, the deprecations are "bad". So being red is a good thing.

I believe the typical case it to replicate a `message()` call.

Reply:
13767b7